### PR TITLE
Telemetry metrics for OTel

### PR DESCRIPTION
**Pull request by Qard**
### What does this PR do?

Adds Telemetry Metrics API and uses it to track types of spans created and finished.

### Motivation

We want visibility into OTel adoption so we should record which way spans are created.

### CODE REVIEW

**Review by bengl — APPROVED** [review]

File: packages/dd-trace/src/telemetry/metrics.js [hunk 1]
```diff
@@ -0,0 +1,219 @@
...
+  rate (name, interval, tags) {
+    return this.getMetric('rate', name, tags, interval)
+  }
+
+  toJSON () {
+    const { namespace } = this
+    return {
+      namespace,
+      lib_language: 'nodejs',
+      lib_version: process.version,
```
Comment (iunanua) [inline packages/dd-trace/src/telemetry/metrics.js:178]: Why are you adding `lib_language` and `lib_version` properties to the payload? 
Shouldn't be added as tags? and shouldn't be `version` instead `lib_version`?

According to the doc:
- Common metrics will have the library’s `lib_language` and `version` as `tags.`
- Language specific metrics will have their respective `lib_version` as `tag`

Comment (Qard) [reply packages/dd-trace/src/telemetry/metrics.js:178]: v1 expects them as part of the payload and this is designed for v1 as v2 is not ready yet and this needs to be out before that is ready.

Comment (Qard) [reply packages/dd-trace/src/telemetry/metrics.js]: Ah, yep, fixed now. :)

File: packages/dd-trace/src/opentracing/span.js [hunk 2]
```diff
@@ -20,6 +23,30 @@ const {
 const unfinishedRegistry = createRegistry('unfinished')
 const finishedRegistry = createRegistry('finished')
 
+const otelEnabled = !!process.env.DD_TRACE_OTEL_ENABLED
```
Comment (tlhunter) [inline packages/dd-trace/src/opentracing/span.js:26]: How about `OTEL_ENABLED` since it's a top level constant?

**Review by tlhunter — DISMISSED** [review]

File: packages/dd-trace/src/opentracing/span.js (hunk 2 above)
Comment (Qard) [reply packages/dd-trace/src/opentracing/span.js]: Done. :)

**Review by tlhunter — APPROVED** [review]

**Pull request merged**